### DISCUSSION TITLE
Custom parameter TOML in ASE integration

### DISCUSSION THREAD

**Discussion by ericmates** [tblite/tblite#86] [Q&A]
Hi all,

I was wondering if, similarly to the --param option in `tblite run`, there is a way of specifying a custom parametrization file for single point calculations in the ASE integration. I'm trying to use the GFN1(Si)-xTB parameters without having to recompile.

Thank you for all the support!!

**Comment by awvwgk** [comment]
There are C bindings available for building the data structure, see https://tblite.readthedocs.io/en/latest/api/c.html#table-data-structure, which can be used to load it via the API when constructing a calculator. I haven't implemented an equivalent mechanism in Python to map a dict to the table data structure. This seems like the preferred way to me since it allows parsing in Python and passing the data structure via C to Fortran.

An alternative might be to pass the file to Fortran and read it internally using the same TOML Fortran routines as `tblite run`, however than the intermediate data structure would be completely opaque in Python.

**Reply by omololuakinojo** [reply]
Good day.
Thank you for the nice work with TBLITE.
I wanted to know if you have the Python equivalent of this?
Like the OP mentioned, I would also like to use GFN in ASE but with my own GFN parameters. 
Could you guide me on how to do this (in ASE) without re-compiling the code?

Thank you very much.

Omololu
